### DISCUSSION TITLE
VLC  - ERR: path must end with a slash

### DISCUSSION THREAD

**Discussion by t3therdev** [bluenviron/mediamtx#433] [Questions]
## Which version are you using?

v0.15.4

## Which operating system are you using?

OS

- [ ] Linux
- [X] Windows
- [ ] macOS

## Describe the issue

Setup a simple brand new installation of 15.4. Change all paths to ch500. Set source as an rtsps stream such as rtsps://ipaddress/ch500
path will be non rtsps (source is rtsps)

Connect with VLC:
ERR: path must end with a slash

Connect with ffplay, works fine. along with other clients.

**Comment by aler9** [comment]
Hello, to check this i need the server logs up to the point that VLC tries to read the stream.

**Comment by Enverex** [comment]
Seeing the same thing. If you put in the URL as is, you get...

`ERR: path must end with a slash (mypath)`

So you add a slash, but then you get...

`ERR: invalid path name: can't end with a slash (mypath/)`

Getting mixed messages here.

**Comment by aler9** [comment]
No one provided logs to analyze the issue, but anyway this is probably what is happening:

1. the server tries to establish a connection to the source (the camera), but it can't since the camera can't be reached
2. a client (VLC?) tries to connect to the server, but the server hasn't established a connection to the source yet, and it rejects the client; VLC tries a fallback, unsupported protocol and this generates the error:

```
path must end with a slash
```

Make sure that the server can reach the camera and wait for the log message

```
[rtsp source] ready
```

Regarding the error, a little improvement could be writing it in another way, such as

```
ERR: client is using an unsupported version of RTSP that doesn't use Content-Base
```

**Comment by tprins228** [comment]
What logs do you need and how do I pull them?  I'm having the same " ERR: path must end with a slash" while running the docker image on a raspberry pi and attempting to access via VLC on a Windows machine.

**Comment by aler9** [comment]
Just provide the server logs, that can be saved to a file by adding `logDestinations: [stdout, file]` to the configuration.

**Comment by tprins228** [comment]
attached are logs and config file (in text format)
[rtsp-simple-server.txt](https://github.com/aler9/rtsp-simple-server/files/6609697/rtsp-simple-server.txt)
[rtsp-simple-server.log](https://github.com/aler9/rtsp-simple-server/files/6609685/rtsp-simple-server.log)

**Reply by aler9** [reply]
Hello, in my opinion Gstreamer is not receiving and / or converting the Raspberry camera to RTSP correctly. 

* try to run gst-launch-1.0 manually and check whether there's any error
* check that the rpicamsrc plugin is installed correctly; otherwise, install it
* check that the camera is installed correctly and the video cam be received by the Raspberry Pi.

**Reply by tprins228** [reply]
@aler9 This was occurring when using your docker container.  If I run the app natively, I do not have issues.

**Reply by aler9** [reply]
my Docker image doesn't contain gst-launch-1.0 nor rpicamsrc - if you want to use rtsp-simple-server AND gstreamer in Docker you have to create a custom image like this (this is for ffmpeg, you have to edit the code to install gstreamer and compile rpicamsrc):

```
FROM aler9/rtsp-simple-server AS rtsp

FROM alpine:3.13
RUN apk add --no-cache ffmpeg

COPY --from=rtsp /rtsp-simple-server /
COPY --from=rtsp /rtsp-simple-server.yml /

ENTRYPOINT ["/rtsp-simple-server" ]
```

**Reply by raypanggh** [reply]
This only happens to me if I'm running as a Windows service. When running in interactive mode, it works fine from VLC.

**Comment by ingGhost** [comment]
any update fix to issue ?

**Comment by wazerstar** [comment]
Getting the same, still trying to figure out how to restream my hdhomerun, cannot get it working at all..

logs froms failing vlc play.

```
2023/12/15 01:40:51 INF MediaMTX v1.4.0
2023/12/15 01:40:51 INF configuration loaded from C:\Users\wazer\Desktop\mediamtx\mediamtx.yml
2023/12/15 01:40:51 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2023/12/15 01:40:51 INF [RTMP] listener opened on :1935
2023/12/15 01:40:51 INF [HLS] listener opened on :8888
2023/12/15 01:40:51 INF [WebRTC] listener opened on :8889 (HTTP), :8189 (ICE/UDP)
2023/12/15 01:40:51 INF [SRT] listener opened on :8890 (UDP)
2023/12/15 01:40:52 INF [RTSP] [conn 127.0.0.1:64088] opened
2023/12/15 01:40:52 INF [RTSP] [conn 127.0.0.1:64088] closed: invalid path name: can't end with a slash (ondemand/)
2023/12/15 01:40:56 INF [RTSP] [conn [::1]:64091] opened
2023/12/15 01:40:56 INF [RTSP] [session 5eb97cbf] created by [::1]:64091
2023/12/15 01:40:56 INF [path ondemand] runOnDemand command started
2023/12/15 01:40:56 INF [path ondemand] runOnDemand command exited: command exited with code 0
2023/12/15 01:41:01 INF [path ondemand] runOnDemand command exited: command exited with code 0
2023/12/15 01:41:06 INF [path ondemand] runOnDemand command stopped: timed out
2023/12/15 01:41:06 INF [RTSP] [conn [::1]:64091] closed: source of path 'ondemand' has timed out
2023/12/15 01:41:06 INF [RTSP] [session 5eb97cbf] destroyed: not in use
2023/12/15 01:41:06 INF [RTSP] [conn [::1]:64092] opened
2023/12/15 01:41:06 INF [RTSP] [conn [::1]:64092] closed: EOF
2023/12/15 01:41:14 INF [RTSP] [conn 127.0.0.1:64093] opened
2023/12/15 01:41:14 INF [path ondemand] runOnDemand command started
2023/12/15 01:41:14 INF [path ondemand] runOnDemand command exited: command exited with code 0
2023/12/15 01:41:19 INF [RTSP] [conn [::1]:64094] opened
2023/12/15 01:41:19 INF [RTSP] [session 5ed67ce3] created by [::1]:64094
2023/12/15 01:41:19 INF [RTSP] [conn [::1]:64094] closed: path of a SETUP request must end with a slash. This typically happens when VLC fails a request, and then switches to an unsupported RTSP dialect
2023/12/15 01:41:19 INF [RTSP] [session 5ed67ce3] destroyed: not in use
2023/12/15 01:41:19 INF [RTSP] [conn [::1]:64095] opened
2023/12/15 01:41:19 INF [RTSP] [conn [::1]:64095] closed: EOF
2023/12/15 01:41:19 INF [path ondemand] runOnDemand command exited: command exited with code 0
2023/12/15 01:41:24 INF [path ondemand] runOnDemand command stopped: timed out
2023/12/15 01:41:24 INF [RTSP] [conn 127.0.0.1:64093] closed: source of path 'ondemand' has timed out
```

**Reply by wazerstar** [reply]
@aler9

**Reply by aler9** [reply]
this doesn't have anything to do with VLC, the issue here is that your `runOnDemand` command fails to start, as clearly written in the logs.

**Comment by aler9** [comment]
this issue has become a list of unrelated problems, i have to lock it.
